File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/operation/mutate/RecordMutateDtoDao.java
package ru.citeck.ecos.records3.record.operation.mutate;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import ru.citeck.ecos.records3.RecordRef;

public interface RecordMutateDtoDao<T> {

    @Nullable
    T getRecToMutate(@NotNull RecordRef recordRef);

    void saveMutatedRec(@NotNull T record);
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/dao/mutate/RecordMutateDtoDao.kt
package ru.citeck.ecos.records3.record.dao.mutate

import ru.citeck.ecos.records3.record.dao.RecordsDao
import kotlin.jvm.Throws

interface RecordMutateDtoDao<T> : RecordsDao {

    /**
     * @param recordId identifier of mutated record
     */
    @Throws(Exception::class)
    fun getRecToMutate(recordId: String): T

    /**
     * @return identifier of mutated record.
     *         Can be different value than recordId in getRecToMutate argument
     */
    @Throws(Exception::class)
    fun saveMutatedRec(record: T): String
}
